docs/devguide.rst

```diff
@@ -61,22 +61,66 @@ is open to interpretation.
   unittest directly or write tests in an object-oriented style.
 * Test generators make more dots and the dots must flow!
 
+You can easily check for style issues, including some outright bugs such
+as mispelled variable names, using pylint. If you're using Anaconda you'll
+need to run "conda install pylint" once. You can easily run pylint on
+the edited files in your uncommited git change:
+
+    $ make lint
+
+If you want to lint the entire code base run:
+
+    $ make lint-all
+
 How to Test
 ================
-First, install nose: http://nose.readthedocs.org/en/latest/
+First, install nose: http://nose.readthedocs.org/en/latest/. Second, ensure
+your cwd is the root directory of the project (i.e., the one containing the
+.git directory).
 
 To perform all unit tests::
 
-    $ cd tests/
-    $ nosetests
+    $ make test-all
+
+Or, if you want to do it the hard way:
+
+    $ scripts/run-tests.xsh
+
+Or, if you want to do it the really hard way:
+
+    $ python3 -c 'import setup; setup.build_tables()'
+    $ XONSHRC='' nosetests
+
+If you're working on a change and haven't yet commited it you can run the
+associated tests with:
 
-This will recursively look through the currently directory, open up every file
-named test_* and run every function (or method) named test_*.
+    $ make test
 
-Nosetests can also take file(s) as an argument. For example, to run just the
-lexer and parser module tests::
+If you want to run specific tests you can specify the test names. For example
+to run test_parser:
 
-    $ nosetests test_lexer.py test_parser.py
+    $ python3 -c 'import setup; setup.build_tables()'
```

Comment:
> Same here, make should be a first-class alternative, though not a requirement.

Make is never a requirement. The contents of `Makefile` can always be executed by hand. Make is simply a well established mechanism in the UNIX community for automating such matters.